### PR TITLE
Build: use gvisor for projects using build.jobs

### PR DESCRIPTION
**Pull request by stsewd**
Closes https://github.com/readthedocs/readthedocs.org/issues/9103

### CODE REVIEW

**Review by ericholscher — APPROVED** [review]
This is a simple implementation. @agjohnson Thoughts?

**Review by agjohnson — COMMENTED** [review]
This makes me a bit less nervous about `build.jobs` in the wild :+1:

We'll need to eventually raise this as a top level setting, perhaps as we remove the feature flag after a test or two. But this is helpful for now, in addition to any projects we want to flag manually.

File: readthedocs/doc_builder/director.py [hunk 1]
```diff
@@ -107,12 +107,14 @@ def create_vcs_environment(self):
         )
 
     def create_build_environment(self):
+        use_gvisor = self.data.config.using_build_tools and self.data.config.build.jobs
```
[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/doc_builder/director.py:110]: it means that the config is using build.tools (and not the build.image setting), this is so we can access config.build.jobs without an attribute error

[the remark this answers is not in the document]
Comment (agjohnson) [reply readthedocs/doc_builder/director.py:110]: Huh, maybe I'm missing part of the `build.jobs` implementation -- is usage of `build.jobs` dependent on usage of `build.tools`?

That is, it seems this config file would not get gvisor in this PR:

```yaml
build:
  jobs:
    pre_install:
      - cowsay meow
```

[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/doc_builder/director.py:110]: That config file isn't allowed, you need to use `build.os` with at least one tool on `build.tools.
https://github.com/readthedocs/readthedocs.org/blob/b186891df4018b0caa751563c377b81f8f9dc869/readthedocs/config/config.py#L758-L758

[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/doc_builder/director.py:110]: basically we have 2 `build` configs, the old one and the one with `build.os/tools`

[the remark this answers is not in the document]
Comment (agjohnson) [reply readthedocs/doc_builder/director.py:110]: Odd, I wouldn't have guessed that the two need to be related. Your implementation at least matches that for now though.

Users might find that requirement a little counterintuitive though, but we can resolve that later.

**Review by agjohnson — APPROVED** [review]

**Comment by ericholscher** [comment]
> Users might find that requirement a little counterintuitive though, but we can resolve that later.

Yea, @humitos & I just discussed this on our call:

Screen Shot 2022-04-21 at 11 09 15 AM

Definitely something where we need to better guide users.

**Pull request merged**